lib/tether/ops_linux.go

```diff
@@ -986,3 +1014,35 @@ func readDir(dir string) ([]os.FileInfo, error) {
 
 	return result, nil
 }
+
+func bindMount(src, target string) error {
+	// no need to return if unmount fails; it's possible that the target is not mounted previously
+	log.Infof("unmounting %s", target)
+	if err := Sys.Syscall.Unmount(target, syscall.MNT_DETACH); err != nil {
+		log.Errorf("failed to unmount %s: %s", target, err)
+	}
+
+	// bind mount src to target
+	log.Infof("bind-mounting %s on %s", src, target)
+	if err := Sys.Syscall.Mount(src, target, ext4FileSystemType, syscall.MS_BIND, ""); err != nil {
```

Comment:
I _think_ that MS_BIND ignores fstype and you coud be less presumptive here:
https://github.com/mindreframer/golang-devops-stuff/blob/master/src/github.com/dotcloud/docker/vendor/src/github.com/docker/libcontainer/mount/init.go#L40